From c05959710efa5515837a44985a2cba77c9cf8617 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Fri, 9 Jan 2026 16:44:35 +0100
Subject: [PATCH] fix: Disable colored tracebacks in test_import_py_err for
 Python 3.14+

Python 3.14 enables colored tracebacks by default, which adds ANSI escape
codes to error messages. This broke the test that checks stderr for a
specific error message. Setting PYTHON_COLORS=0 disables this behavior.
---
 Gaudi/tests/pytest/test_import_py_err.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Gaudi/tests/pytest/test_import_py_err.py b/Gaudi/tests/pytest/test_import_py_err.py
index 33e66511ba..80fcd9f677 100644
--- a/Gaudi/tests/pytest/test_import_py_err.py
+++ b/Gaudi/tests/pytest/test_import_py_err.py
@@ -1,5 +1,5 @@
 #####################################################################################
-# (c) Copyright 2024 CERN for the benefit of the LHCb and ATLAS collaborations      #
+# (c) Copyright 2024-2026 CERN for the benefit of the LHCb and ATLAS collaborations #
 #                                                                                   #
 # This software is distributed under the terms of the Apache version 2 licence,     #
 # copied verbatim in the file "LICENSE".                                            #
@@ -16,6 +16,8 @@ from GaudiTesting import GaudiExeTest
 class Test(GaudiExeTest):
     environment = [
         f"JOBOPTSEARCHPATH={Path(__file__).parent.parent / 'pyjobopts'}",
+        # Disable colored tracebacks in Python 3.14+ to get predictable output
+        "PYTHON_COLORS=0",
     ]
 
     command = ["gaudirun.py", "--dry-run", "--verbose", "main_err.py"]
-- 
GitLab

